src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -475,11 +507,14 @@ void FancyZones::AddZoneWindow(HMONITOR monitor, PCWSTR deviceId) noexcept
     wil::unique_cotaskmem_string virtualDesktopId;
     if (SUCCEEDED_LOG(StringFromCLSID(m_currentVirtualDesktopId, &virtualDesktopId)))
     {
-        const bool flash = m_settings->GetSettings().zoneSetChange_flashZones;
+        const bool newVirtualDesktop = m_virtualDesktopIds[m_currentVirtualDesktopId];

```

Comment:
`unordered_*`'s `operator[]`S insert a value if it wasn't present on access, so let's use find(...) != end instead, since we explicitly assign `false` later on.